Ports/VVVVVV/patches/0002-Add-SerenityOS-in-platform-specific-defines.patch

```diff
@@ -0,0 +1,54 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Grigoris Pavlakis <grigpavl@ece.auth.gr>
+Date: Thu, 9 Jun 2022 00:21:51 +0300
+Subject: [PATCH] Add SerenityOS in platform-specific defines
+
+---
+ desktop_version/src/FileSystemUtils.cpp | 6 +++---
+ third_party/tinyxml2/tinyxml2.cpp       | 2 +-
+ 2 files changed, 4 insertions(+), 4 deletions(-)
+
+diff --git a/desktop_version/src/FileSystemUtils.cpp b/desktop_version/src/FileSystemUtils.cpp
+index 833a3de..d30f6b5 100644
+--- a/desktop_version/src/FileSystemUtils.cpp
++++ b/desktop_version/src/FileSystemUtils.cpp
+@@ -26,7 +26,7 @@ int mkdir(char* path, int mode)
+ 	return CreateDirectoryW(utf16_path, NULL);
+ }
+ #define VNEEDS_MIGRATION (mkdirResult != 0)
+-#elif defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__)
++#elif defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__) || defined (__serenity__)
+ #include <unistd.h>
+ #include <dirent.h>
+ #include <limits.h>
+@@ -1005,7 +1005,7 @@ static void PLATFORM_migrateSaveData(char* output)
+ 	char oldLocation[MAX_PATH];
+ 	char newLocation[MAX_PATH];
+ 	char oldDirectory[MAX_PATH];
+-#if defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__)
++#if defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__) || defined (__serenity__)
+ 	DIR *dir = NULL;
+ 	struct dirent *de = NULL;
+ 	DIR *subDir = NULL;
+@@ -1018,7 +1018,7 @@ static void PLATFORM_migrateSaveData(char* output)
+ 		return;
+ 	}
+ 	const char oldPath[] =
+- #if defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__)
++ #if defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__) || defined(__serenity__)
+ 		"/.vvvvvv/";
+  #elif defined(__APPLE__)
+ 		"/Documents/VVVVVV/";
+diff --git a/third_party/tinyxml2/tinyxml2.cpp b/third_party/tinyxml2/tinyxml2.cpp
+index 25b18ed..dba42e8 100644
+--- a/third_party/tinyxml2/tinyxml2.cpp
++++ b/third_party/tinyxml2/tinyxml2.cpp
+@@ -103,7 +103,7 @@ distribution.
+ #if defined(_WIN64)
+ 	#define TIXML_FSEEK _fseeki64
+ 	#define TIXML_FTELL _ftelli64
+-#elif defined(__APPLE__) || (__FreeBSD__) || (__OpenBSD__)
++#elif defined(__APPLE__) || (__FreeBSD__) || (__OpenBSD__) || (__serenity__)
```

Comment:
I know that the `defined` is missing from the other entries as well, but does this even have any effect without it?

Reply:
Curiously, yes it has an effect. If I delete `(__serenity__)` I'm getting `error: 'fseeko64' was not declared in this scope; did you mean 'fseeko'?`. I can do the defined though if you need it

Reply:
Then it's fine, I just wanted to make sure that we don't end up in the wrong branch by accident.